src/Compilers/Core/Portable/Compilation/OptimizationLevel.cs

```diff
@@ -48,42 +48,46 @@ public enum OptimizationLevel
 
     internal static class OptimizationLevelFacts
     {
+        internal static (OptimizationLevel OptimizationLevel, bool DebugPlus) DefaultValues { get; } = (OptimizationLevel.Debug, false);
+
         public static string ToPdbSerializedString(this OptimizationLevel optimization, bool debugPlusMode)
-        => optimization switch
-        {
-            OptimizationLevel.Release => "release",
-            OptimizationLevel.Debug => debugPlusMode ? "debug-plus" : "debug",
-            _ => throw ExceptionUtilities.UnexpectedValue(optimization)
-        };
+            => (optimization, debugPlusMode) switch
+            {
+                (OptimizationLevel.Release, true) => "release-debug-plus",
+                (OptimizationLevel.Release, false) => "release",
+                (OptimizationLevel.Debug, true) => "debug-plus",
+                (OptimizationLevel.Debug, false) => "debug",
+                _ => throw ExceptionUtilities.UnexpectedValue(optimization)
+            };
 
         public static bool TryParsePdbSerializedString(string value, out OptimizationLevel optimizationLevel, out bool debugPlusMode)
         {
-            if (value == "release")
-            {
-                optimizationLevel = OptimizationLevel.Release;
-                debugPlusMode = false;
-                return true;
-            }
-            else if (value == "debug")
+            switch (value)
             {
-                optimizationLevel = OptimizationLevel.Debug;
-                debugPlusMode = false;
-                return true;
+                case "release-debug-plus":
```

Comment:
>release-debug-plus [](start = 22, length = 18)

Is this an actual setting that we allow? I thoght we have 3 optimization "kinds".

Reply:
Yes it's fully supported. This didn't exist in Roslyn 1.0 and caused fairly massive regressions in the user experience for azure scenarios. It was added back in 1.1 or 1.2 (can't remember). Not properly supporting this is what is causing our unit tests to not round trip correctly.